### PR TITLE
VM: Add test to ensure that IDENTITY precompile copies the memory (chain split from 2021-08)

### PR DESCRIPTION
**Pull request by jochem-brouwer**
The details of the exploit have been shared on #allcoredevs so we are OK to also publicly create this test.

We are not susceptible.

This PR replays most parts of the tx in a sandbox environment, the only thing we should care about is the actually deployed code, which matches what happens on chain.

### CODE REVIEW

File: packages/vm/tests/api/runCall.spec.ts [hunk 1]
```diff
@@ -223,3 +223,40 @@ tape('Ensure that Istanbul sstoreCleanRefundEIP2200 gas is applied correctly', a
...
 })
+
+tape('Ensure that IDENTITY precompile copies the memory', async (t) => {
+  // This test replays the EVM exploit; https://etherscan.io/tx/0x1cb6fb36633d270edefc04d048145b4298e67b8aa82a9e5ec4aa1435dd770ce4
+  // setup the accounts for this test
+  const caller = new Address(Buffer.from('1a02a619e51cc5f8a2a61d2a60f6c80476ee8ead', 'hex')) // caller addres
+  // setup the vm
+  const common = new Common({ chain: Chain.Mainnet, hardfork: Hardfork.London })
+  const vm = new VM({ common: common })
+  const code = '3034526020600760203460045afa602034343e604034f3'
```
[the remark this answers is not in the document]
Comment (ryanio) [reply packages/vm/tests/api/runCall.spec.ts:234]: adding a link to a technical writeup would be sufficient too, just something to learn more about what's going on.

[the remark this answers is not in the document]
Comment (jochem-brouwer) [reply packages/vm/tests/api/runCall.spec.ts:234]: I think we can wait until Geth releases their security vulnerability report, we can link that then :+1:

[the remark this answers is not in the document]
Comment (holgerd77) [reply packages/vm/tests/api/runCall.spec.ts:234]: "the EVM exploit"

Also, could you make this more generic (no one will know in a year from now what *the* EVM exploit will be. 😋) So rather something like "an EVM exploit from 2021-08-...`.

[the remark this answers is not in the document]
Comment (jochem-brouwer) [reply packages/vm/tests/api/runCall.spec.ts:234]: Yes, good point. Let's wait until Geth publishes their disclosure, then I'll make this description a bit more descriptive. Note that this is not a complete test (on the entire state trie) but that seems to much to me - this should cover what went wrong, and we'll probably see something similar back in `ethereum/tests` later on anyways)

[the remark this answers is not in the document]
Comment (holgerd77) [reply packages/vm/tests/api/runCall.spec.ts:234]: Added a "WIP" as well as a "Blocked" label (due to this "Let's wait..." comment) to this PR. Please correct if this is not fitting the needs.

[the remark this answers is not in the document]
Comment (jochem-brouwer) [reply packages/vm/tests/api/runCall.spec.ts:236]: Updated the comment with a link + permalink to the post-mortem writeup.

**Review by holgerd77 — APPROVED** [review]
LGTM

**Pull request merged**